lib/webhooks/process.ts

```diff
@@ -1,25 +1,43 @@
 import {StatusCode} from '@shopify/network';
 
 import {
-  abstractConvertRequest,
   abstractConvertResponse,
   AdapterResponse,
-  getHeader,
-  Headers,
   isOK,
-  NormalizedRequest,
   NormalizedResponse,
 } from '../../runtime/http';
-import {createSHA256HMAC} from '../../runtime/crypto';
-import {HashFormat} from '../../runtime/crypto/types';
-import {ShopifyHeader} from '../types';
 import {ConfigInterface} from '../base-types';
-import {safeCompare} from '../auth/oauth/safe-compare';
 import * as ShopifyErrors from '../error';
 import {logger} from '../logger';
 
-import {WebhookRegistry, WebhookProcessParams, DeliveryMethod} from './types';
-import {topicForStorage} from './registry';
+import {
+  WebhookRegistry,
+  WebhookProcessParams,
+  DeliveryMethod,
+  WebhookValidationErrorReason,
+  WebhookValidationMissingHeaders,
+  WebhookValidationValid,
+  WebhookValidationInvalid,
+} from './types';
+import {validate} from './validate';
+
+interface HandlerCallResult {
```

Comment:
Should `HandlerCallResult` and `ErrorCallResult` be 'merged' into one unique `CallResult` since they have the structure?